EDP-380C-28-R2-Lab3.R

```diff
@@ -12,3 +12,527 @@ source("scripts/rmvnorm.R")
+# b0    42.3863641  4.37899519903836   9.67947260714761  1.9659287164877e-10
+# b1    -3.3920819 0.820802491744326  -4.13264087599804 0.000294156959036295
+# b2    -1.5280010 0.419753253251725  -3.64023630377079  0.00109260914427508
+# b3    -0.5228629 0.778870339652137 -0.671309355170919    0.507524370714282
+# SD(e)  2.5921848                NA                 NA                   NA
+# R2     0.8182681                NA                 NA                   NA
+
+
+## 3.a
+set.seed(21389)
+# Predefine parameters 
+param_list <- list(Rho = 0.3,
+                   t_mu = c(5, 10),
+                   t_sigma = c(1, 2))
+
+# Function that translates parameter list into rmvnorm() inputs 
+param2rmvnorm_list <- function(param_list) {
+  
+  # Create R_x & change diag
+  R_x <- matrix(param_list$Rho,
+                dim(diag(c(param_list$t_sigma)))[1],
+                dim(diag(c(param_list$t_sigma)))[2])
+  diag(R_x) <- 1
+  
+  # Create list
+  rmvnorm_list <- list(
+    ## Mean vector
+    mu = matrix(param_list$t_mu, 
+                ncol = 1), 
+    ## Sigma matrix 
+    Sigma = diag(c(param_list$t_sigma)) %*% 
+      R_x %*% 
+      diag(c(param_list$t_sigma))
+  )
+  
+  return(rmvnorm_list)
+}
+
+# Change parameter list to rmvnorm inputs
+pred_rmvnorm_list <- param2rmvnorm_list(param_list)
+
+# Generate predictor data 
+pred_data <- rmvnorm(n = 100000, 
+        mu = pred_rmvnorm_list$mu,
+        Sigma = pred_rmvnorm_list$Sigma)
+
+
+## 3.b
+set.seed(23921)
+# Predefine outcome parameters
+outcome_param_list <- list(beta1 = 1,
+                           beta2 = 1, 
+                           R2 = 0.6,
+                           mu_y = 10, 
+                           sd_y = 5)
+
+# Betas
+Betas_3b <- matrix(as.double(
+  c(outcome_param_list[startsWith(names(outcome_param_list), 
+                                  prefix = "beta")]))
+)
+
+
+# Compute SD error for Y
+SD_e <- sqrt(
+  t( Betas_3b ) %*% 
+    pred_rmvnorm_list$Sigma %*% 
+    Betas_3b %*% 
+    ( (1/outcome_param_list$R2) - 1)
+)
+
+# Find & set intercept
+intercept_3b <- outcome_param_list$mu_y - 
+  t( pred_rmvnorm_list$mu ) %*% Betas_3b # intercept = -5
+
+# Generate Y data
+outcome_data <- rnorm(n = 100000, 
+      mean = as.vector(intercept_3b) + pred_data %*% Betas_3b,
+      sd = SD_e)
+
+# Answer 
+
+## Analyze with OLS function 
+OLSfunc(outcome_data, pred_data)
+
+## Difference for mean Y parameter 
+outcome_param_list$mu_y - mean(outcome_data) 
+## Difference for SD Y parameter 
+outcome_param_list$sd_y - sd(outcome_data)
+
+# Estimate                  SE           t value  p
+# b0    -4.9370600  0.0404634289861931 -122.012892517685  0
+# b1     0.9891621 0.00677154296619353  146.076328907351  0
+# b2     0.9997359 0.00337368817415204  296.333224140717  0
+# SD(e)  2.0361731                  NA                NA NA
+# R2     0.5978875                  NA                NA NA
+
+# ## Difference for mean Y parameter
```

Comment:
Look at the difference between **each** parameter. Create a table like the one below.

```r
#            pop        est           dif
# b0    -5.00000
# b1     1.00000 
# b2     1.00000  
# SD(e)  2.03306  
# R2     0.60000  
```